democracy_club/apps/core/s3_lambda_storage.py

```diff
@@ -1,16 +1,66 @@
+import mimetypes
 import os
 from distutils.dir_util import copy_tree
 
 from pipeline.compilers.sass import SASSCompiler
 
 from django.conf import settings
+from django.core.files.base import File
 from storages.backends.s3boto3 import S3Boto3Storage
 from django.contrib.staticfiles.storage import ManifestFilesMixin
 from pipeline.storage import PipelineMixin
 
 from django.contrib.staticfiles.finders import FileSystemFinder
 
 
+class PatchedS3Boto3Storage(S3Boto3Storage):
+
+    """
+    Note: We need to patch S3Boto3Storage
+    to apply a fix which stops botocore from throwing
+
+    Invalid type for parameter ContentType, value: b'text/javascript',
+    type: <class 'bytes'>, valid types: <class 'str'>: ParamValidationError
+
+    Unfortunately this lives in the middle of quite a long method
+    so we've got a big old chunk of copy+pasted code here :(
+
+    Also we need to be really careful we don't upgrade the
+    django-storages package while this monkey-patch is in place
+    as this may yield unexpected behaviour.
```

Comment:
Could you grab the version from storages and raise if it's not equal to the one we're pinned to? Will save us forgetting about this and upgrading (or at least would make something fail quickly)

Reply:
Yep - good call. Since The first line in the deploy script runs `manage.py check`, I've [implemented it as a system check](https://github.com/DemocracyClub/Website/pull/117/commits/47b665c442864d5442529e3a54072d31858a8bf6)